src/test/scala/chiseltest/tests/ElementTest.scala

```diff
@@ -125,4 +125,42 @@ class ElementTest extends FlatSpec with ChiselScalatestTester {
       c.io.expect(31.75.F(2.BP), 0.25.F(2.BP), -32.F(2.BP))
     }
   }
+
+  it should "work with Interval" in {
+
+    val inputRange = range"[-6, 6].2"
+    val outputRange = range"[-4.5,4.5].2"
+
+    test(new Module {
+      val io = IO(new Bundle {
+        val input = Input(Interval(inputRange))
+
+        val out1 = Output(Interval(inputRange))
+        val out1Clip = Output(Interval(outputRange))
+        val out1Squeeze = Output(Interval(outputRange))
+        val out1Wrap = Output(Interval(outputRange))
+      })
+      io.out1 := io.input
+      io.out1Clip := io.input.clip(io.out1Clip)
```

Comment:
This looks slightly suspect. I assume what this means is io.out1Clip := io.input clipped to the interval defined by io.out1Clip, where clip ignores the hardware value of the argument? In that case, would it be cleaner to make that drop-hardware-value explicit with a chiselTypeOf operation, or is that value still somehow necessary?

Reply:
Resolved in offline discussion: appears the hardware node is needed by FIRRTL to get the type (in FIRRTL) of the intervals. So not a problem to be solved here, though might be worth revisiting in chisel.